### PR TITLE
CKBox plugins should not wait for token fetch.

### PR DESCRIPTION
**Pull request by Mati365**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Other (ckbox): The plugin no longer slows down the editor startup because it fetches the token in the background instead of during the editor’s initialization. Closes https://github.com/ckeditor/ckeditor5/issues/16760

MINOR BREAKING CHANGE (ckbox): The `CKBoxUtils#getWorkspaceId` and `CKBoxUtils#getToken` methods now return a promise instead of a resolved value.

### CODE REVIEW

**Review by illia-stv — APPROVED** [review]

File: packages/ckeditor5-ckbox/tests/ckboxuploadadapter.js [hunk 1]
```diff
@@ -1137,9 +1137,9 @@ describe( 'CKBoxUploadAdapter', () => {
 						.then( () => {
 							throw new Error( 'Expected to be rejected.' );
 						}, err => {
-							expect( console.error.callCount ).to.equal( 1 );
+							expect( console.error.callCount ).to.equal( 2 );
```
Comment (niegowski) [inline packages/ckeditor5-ckbox/tests/ckboxuploadadapter.js:1140]: Why has this changed?

File: packages/ckeditor5-ckbox/tests/ckboxuploadadapter.js [hunk 2]
```diff
@@ -1137,9 +1137,9 @@ describe( 'CKBoxUploadAdapter', () => {
...
 							expect( console.error.firstCall.args[ 0 ] ).to.match( /^ckbox-access-default-workspace-error/ );
-							expect( err ).to.equal( 'Cannot access default workspace.' );
+							expect( err ).to.equal( 'Cannot determine a category for the uploaded file.' );
```
Comment (niegowski) [inline packages/ckeditor5-ckbox/tests/ckboxuploadadapter.js:1142]: Why has this changed?

**Review by niegowski — COMMENTED** [review]
I'd suggest adding a test for the original issue to make sure there is no regression in the future.

File: packages/ckeditor5-ckbox/tests/ckboxuploadadapter.js (hunk 2 above)
Comment (Mati365) [reply packages/ckeditor5-ckbox/tests/ckboxuploadadapter.js:1142]: Hm, good question. I see that it should not be changed. but I think that `await this._token` might result in forwarding incorrect exception.

File: packages/ckeditor5-ckbox/tests/ckboxuploadadapter.js (hunk 1 above)
Comment (Mati365) [reply packages/ckeditor5-ckbox/tests/ckboxuploadadapter.js]: Reverted.

File: packages/ckeditor5-ckbox/tests/ckboxuploadadapter.js (hunk 2 above)
Comment (Mati365) [reply packages/ckeditor5-ckbox/tests/ckboxuploadadapter.js]: @niegowski I found an issue. The problem was that the exception should have been thrown before the `try {` block in `_getAvailableCategories`. After the changes, it was thrown inside the block, causing the exception to be caught in the wrong place.

**Comment by Mati365** [comment]
> I'd suggest adding a test for the original issue to make sure there is no regression in the future.

Ok 👍🏻

**Review by niegowski — APPROVED** [review]

**Comment by Mati365** [comment]
Asked QA for testing. @arkflpc Can you take a look?

**Review by arkflpc — APPROVED** [review]

**Comment by charlttsie** [comment]
We have tested the PR with @marcelmroz and @kubaklatt and can confirm that the change noticeably reduces the editor's loading time when when fetching the token is delayed. At the same time, we haven't found any regressions, so the PR looks good to us 👍

**Comment by Mati365** [comment]
@charlttsie Cool! Thanks!

**Pull request merged**